sed: fix faulty example and add remove line from file osx (#4781)
tldr-pages/tldr: pages/osx/sed.md

`sed -E 's/{{regex}}/{{replace}}/g' {{filename}}`

- Replace all occurrences of a string [i]n a file, overwriting the file (i.e. in-place):

> `sed -i '' 's/{{find}}/{{replace}}/g' {{filename}}`

Reviewer: We generally prefer long-form options in tldr pages for clarity.

Reviewer 2: See https://github.com/tldr-pages/tldr/pull/4781#discussion_r510128984

Reviewer 2: @sbrl Please check https://github.com/tldr-pages/tldr/pull/4781#discussion_r510128984

Reviewer: Oops! I must have missed this. Ignore this suggestion.

> - Replace all occurrences of a string in a file, overwriting the file (i.e. in-place):

Reviewer 2: I'm keeping the ` (i.e. in-place)` because is in the Linux page too, but I don't like it.
Please highlight the `[i]` to get a meaning to that letter:
```suggestion
- Replace all occurrences of a string [i]n a file, overwriting the file (i.e. in-place):
```

- Replace separator / by any other character not used in the find or replace patterns, e.g., #:

`sed 's#{{find}}#{{replace}}#' {{filename}}`

> - Remove line number 123 from a file, interactively with no backup:

Reviewer 2: Do the command ask for confirmation? I want to be sure: Do this delete a line, or a number in the line?
```suggestion
- Remove a line from a file, prompting for confirmation:
```

Author: Hi @navarroaxel!
This command does NOT ask for confirmation and will delete the line matching that number. I've updated the PR, is this more clear?

Author: Resolved in a later convo

> - Delete line number 123 from a file, in-place without saving a backup:

Reviewer 2: Based on the example 3, and we can use this highlight `[]` to explain what `d` means:
```suggestion
- [d]elete the line at the specific line number in a file, overwriting the file:
```

> `sed -i '' '123d' {{filename}}`

Reviewer 2: What's `d` means?

Author: `d` is for delete

Reviewer 2: What about this: https://github.com/tldr-pages/tldr/pull/4781/files#r510070349:
```suggestion
`sed --in-place='' '{{line_number}}d' {{filename}}`
```

Reviewer 2: @stefanwb wdyt about this?

Author: Being more explicit is better indeed 👍

Author: How ever this is not a valid option on osx...
```❯ sed --in-place
sed: illegal option -- -
usage: sed script [-Ealn] [-i extension] [file ...]
       sed [-Ealn] [-i extension] [-e script] ... [-f script_file] ... [file ...]
```

Reviewer 2: And why the example 3 use `--in-place'?

Author: That's faulty, it's not supported and should be removed here

Reviewer 2: Oh, yeah I read your original PR title, why you didn't send the deletion/edition of that example then?

Author: I've updated both in a new commit

> `sed --in-place='' '{{line_number}}d' {{filename}}`

Author: ```suggestion
`sed -i '' '{{line_number}}d' {{filename}}`
```

Author: How about this?
